client/signup/steps/social-profiles/index.tsx

```diff
@@ -1,53 +1,63 @@
-import 'calypso/state/difm/init';
 import { useTranslate } from 'i18n-calypso';
 import { useEffect } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import { useDispatch } from 'react-redux';
 import startingPointImageUrl from 'calypso/assets/images/onboarding/starting-point.svg';
 import StepWrapper from 'calypso/signup/step-wrapper';
-import {
-	resetSocialProfiles,
-	updateSocialProfiles,
-} from 'calypso/state/difm/social-profiles/actions';
-import { getSocialProfiles } from 'calypso/state/difm/social-profiles/selectors';
 import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
 import SocialProfiles from './social-profiles';
-import type { SocialProfilesState } from 'calypso/state/difm/social-profiles/schema';
+import type { SocialProfilesState } from './types';
 
 import './style.scss';
 
 interface Props {
 	goToNextStep: () => void;
-	submitSignupStep: ( { stepName, wasSkipped }: { stepName: string; wasSkipped: boolean } ) => void;
-	goToStep: ( stepName: string ) => void;
 	stepName: string;
+	signupDependencies: SocialProfilesState;
 }
 
 export default function SocialProfilesStep( props: Props ): React.ReactNode {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
+	const { stepName, signupDependencies, goToNextStep } = props;
 
 	const headerText = translate( 'Do you have social media profiles?' );
 	const subHeaderText = translate( 'You can change your mind at any time.' );
 
 	useEffect( () => {
-		dispatch( saveSignupStep( { stepName: props.stepName } ) );
-	}, [ dispatch, props.stepName ] );
+		dispatch( saveSignupStep( { stepName: stepName } ) );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Any reason why this was ignored?

Reply:
Yes, we need to dispatch this action exactly once the component is mounted. An empty dependency array will ensure that it runs only once. 🙂 
Example: https://github.com/Automattic/wp-calypso/blob/765723346be8684cd5c225fa151e305275795d4a/client/signup/steps/design-picker/index.jsx#L91-L98